### PR TITLE
Relay: updated with new API version

### PR DESCRIPTION
**Pull request by v-Ajnava**
### Best Practice Checklist
Check these items before submitting a PR... See the Contribution Guide for the full detail: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md 

1. uri's compatible with all clouds (Stack, China, Government)
1. Staged artifacts use _artifactsLocation & _artifactsLocationSasToken
1. Use resourceGroup().location for resource locations
1. Folder names for artifacts (nestedtemplates, scripts, DSC)
1. Use literal values for apiVersion (no variables)
1. Parameter files (GEN-UNIQUE for value generation and no "changemeplease" values
1. $schema and other uris use https
1. Use uniqueString() whenever possible to generate names for resources.  While this is not required, it's one of the most common failure points in a deployment. 
1. Update the metadata.json with the current date

For details: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/bp-checklist.md

- [X] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

Swagger Link: https://github.com/Azure/azure-rest-api-specs/blob/current/specification/relay/resource-manager/Microsoft.Relay/2017-04-01/relay.json

### Changelog

* Updated the API version to use 2017-04-01
* added new properties for wcfRelay and Hybridconnection relays

### CODE REVIEW

File: 201-azure-relay-create-all-resources/metadata.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
-    "itemDisplayName": "Create an Azure Relay namespace with all resources",
-    "description": "This template enables you to deploy an Azure Relay namespace with standard SKU, a WCF Relay and a HybridConnection.",
-    "summary": "This template creates an Azure Relay namespace, a WCF Relay and a HybridConnection.",
-    "githubUsername": "grandhiramesh",
-    "dateUpdated": "2016-10-25"
+  "itemDisplayName": "Create an Azure Relay namespace with all resources",
+  "description": "This template enables you to deploy an Azure Relay namespace with standard SKU, a WCF Relay and a HybridConnection.",
+  "summary": "This template creates an Azure Relay namespace, a WCF Relay and a HybridConnection.",
+  "githubUsername": "v-ajnava",
```
Comment (bmoore-msft) [inline 201-azure-relay-create-all-resources/metadata.json]: do you want to take ownership of all of these samples?

File: 301-azure-relay-create-authrule-namespace-and-wcfrelay/azuredeploy.parameters.json [hunk 2]
```diff
@@ -0,0 +1,21 @@
+{
+    "$schema": "http://schema.management.azure.com/schemas/2015-01-01/deploymentParameters.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "namespaceName": {
+            "value": "GEN-UNIQUE-RL-NM-001"
```
Comment (bmoore-msft) [inline 301-azure-relay-create-authrule-namespace-and-wcfrelay/azuredeploy.parameters.json]: None of the GEN values here are valid, if they don't need to be globally unique, just use a defaultValue in the template - if they do need to be unique then use GEN-UNIQUE

File: 301-azure-relay-create-authrule-namespace-and-wcfrelay/azuredeploy.json [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+    "wcfrelayAuthorizationRuleName": {
+      "type": "string",
+      "metadata": {
+        "description": "Name of the WcfRelay AuthorizationRule"
+      }
+    }
+  },
+  "variables": {
+    "location": "[resourceGroup().location]",
+    "apiVersion": "2017-04-01",
```
Comment (bmoore-msft) [inline 301-azure-relay-create-authrule-namespace-and-wcfrelay/azuredeploy.json]: remove this and use a literal value

File: 201-azure-relay-create-all-resources/metadata.json (hunk 1 above)
Comment (v-Ajnava) [reply 201-azure-relay-create-all-resources/metadata.json]: Yes, talked with Team here

File: 301-azure-relay-create-authrule-namespace-and-wcfrelay/azuredeploy.parameters.json (hunk 2 above)
Comment (v-Ajnava) [reply 301-azure-relay-create-authrule-namespace-and-wcfrelay/azuredeploy.parameters.json]: used GEN-UNOUE

File: 301-azure-relay-create-authrule-namespace-and-wcfrelay/azuredeploy.json (hunk 3 above)
Comment (v-Ajnava) [reply 301-azure-relay-create-authrule-namespace-and-wcfrelay/azuredeploy.json]: Actually we have used the similar way in the Service Bus and EventHub.

Comment (bmoore-msft) [reply 301-azure-relay-create-authrule-namespace-and-wcfrelay/azuredeploy.json]: It's not optional, it prevents schema validation from being done.

Comment (v-Ajnava) [reply 301-azure-relay-create-authrule-namespace-and-wcfrelay/azuredeploy.json]: resolved.

**Comment by v-Ajnava** [comment]
@bmoore-msft, Have resolved the review comments, can you please review and let me know your feedback.

**Pull request merged**